### ISSUE TITLE
Error: This expression has type bytes but an expression was expected of type string

### ISSUE THREAD

**Issue by orbifx** [janestreet/sexplib#31]
I'm trying to build sexplib on 4.06.0+trunk on ARM for some tests.

Getting this error:
```
#       ocamlc src0/sexplib0__Sexp.{cmo,cmt} (exit 2)
# (cd _build/default && /home/fox/.opam/4.06.0+trunk/bin/ocamlc.opt -w -40 -g -bin-annot -no-alias-deps -I src0 -open Sexplib0 -o src0/sexplib0__Sexp.cmo -c -impl src0/sexp.ml)
# File "src0/sexp.ml", line 77, characters 6-8:
# Error: This expression has type bytes but an expression was expected of type
#          string
#     ocamlopt src0/sexplib0__Sexp.{cmx,o} (exit 2)
# (cd _build/default && /home/fox/.opam/4.06.0+trunk/bin/ocamlopt.opt -w -40 -g -no-alias-deps -I src0 -open Sexplib0 -o src0/sexplib0__Sexp.cmx -c -impl src0/sexp.ml)
# File "src0/sexp.ml", line 77, characters 6-8:
# Error: This expression has type bytes but an expression was expected of type
#          string
```

Has sexplib not been adapted since the change for `string`?

**Comment by mmottl** [comment]
This problem apparently affects everyone using `base`, in particular many `jbuilder` configuration discovery files.  A large number of packages hence won't be able to build.  It would be great if this can be fixed before the release of OCaml 4.06.

**Comment by xclerc** [comment]
It should be fixed in the `v0.9` branch; an opam PR has also
been opened: https://github.com/ocaml/opam-repository/pull/10534

**Comment by hhugo** [comment]
sexplib.v0.9.3 is available in opam and compatible with OCaml 4.06
